### ISSUE TITLE
Replace debugger visualizer with a Visual Studio extension

### ISSUE THREAD

**Issue by snikolayev** [NRules/NRules#112]
Create a Visual Studio extension that would allow visualization of rule sessions, facts within nodes. Long term potentially extending into rule debugging, etc.

**Comment by andregomars** [comment]
still working on it? Or any plan on VS Code extension to allow visualization of rule sessions and etc.?

**Comment by snikolayev** [comment]
Yes, I'm still working on this, but can't say it's going well. I started on this several times and hit a dead-end. Can't find a way to interact with the application being debugged from the extension in a way that gives me what I need. With the debugger visualizer, you can inject code into the debuggee that then interacts with the debugger. With VSIX I cannot find a similar mechanism. The best I found was to evaluate expressions within the debuggee, and they can only return strings, so not too flexible.
And the problem with the debugger visualizer is that it cannot be hosted in a panel within visual studio, so it's very limited in that sense.
But I keep digging. Obviously if you @andregomars or anyone else has experience with this and can offer some suggestions or help that would be greatly appreciated.
